### PR TITLE
Fix bloom filter compatibility issue caused by naming

### PR DESCRIPTION
**Pull request by v01dstar**
https://github.com/tikv/tikv/issues/17272
This is a temp fix for 6.29. Will propose a different fix for 8.10.

Since TiKV only uses RocksDB builtin bloom filters, this should be able to fix the issue.
Also planning to revert the PR that introduced this bug in rust-rocksdb, i.e. remove the suffix like ".FullBloom", "BlockBloom" and "Ribbon". After all, "BlockBloom" has been deprecated in RocksDB newer versions anyway.

### CODE REVIEW

**Comment by v01dstar** [comment]
@hbisheng

File: table/block_based/block_based_table_reader.cc [hunk 1]
```diff
@@ -900,29 +898,29 @@ Status BlockBasedTable::PrefetchIndexAndFilterBlocks(
...
+          std::make_pair(Rep::FilterType::kBlockFilter, kFilterBlockPrefix)}) {
+      auto filter_type = pair.first;
+      Slice prefix = pair.second;
+      meta_iter->Seek(prefix);
+      if (meta_iter->status().ok() && meta_iter->Valid()) {
+        Slice key = meta_iter->key();
+        if (key.starts_with(prefix)) {
+          key.remove_suffix(prefix.size());
+          Slice filter_policy_name_slice = Slice(rep_->filter_policy->Name());
+          if (key.starts_with("rocksdb.BuiltinBloomFilter") ||
```
Comment (Connor1996) [inline table/block_based/block_based_table_reader.cc:921]: please add a comment here

Comment (v01dstar) [reply table/block_based/block_based_table_reader.cc:921]: Done

**Review by overvenus — APPROVED** [review]

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**